model/src/main/proto/exploration.proto

```diff
@@ -322,6 +322,34 @@ message UserAnswer {
   WrittenTranslationContext written_translation_context = 6;
 }
 
+// Corresponds to a raw item selection answer that user has selected.
+message ItemSelectionRawAnswer {
+  repeated int32 selected_indexes = 1;
+}
+
+// Corresponds to a raw representation of the current answer entered by the user which is used to
+// retain state on configuration changes.
+message RawUserAnswer {
+  oneof answer_input_type {
+    // A raw answer entered by user in text input interaction.
+    string text = 1;
+    // A raw answer entered by user in fraction input interaction.
+    string fraction = 2;
```

Comment:
Created a separate field for each interaction because if we keep common field for all text based interaction like previously, then when we open the text-based interactions one after another, previously saved rawUserAnswer is displayed. For example if we have entered answer in numeric interaction and then open ratio interaction, then we get numeric interaction's saved answer inside ratio interaction that's why I have created a separate field for each interaction

Reply:
That sounds like a bug in the implementation, not a problem with the way it's modeled @vrajdesai78. You'll need to make sure that the saved state is properly cleared when a new card is opened. I suspect the bug you described still exists for the case where the same type of interaction occurs twice in a row.

I suggest changing this modeling back and fixing the issue directly, and also adding tests for those cases (i.e. for two different and two of the same interactions not sharing initial state).

Reply:
Solved this issue by introducing a nwe resetRawUserAnswer function which will be called when submitButton is clicked.

Reply:
Resolving this now as as mentioned an previous comment.